anta/result_manager/models.py

```diff
@@ -58,6 +62,9 @@ class TestResult(BaseModel):
     messages: list[str] = []
     custom_field: str | None = None
 
+    # This is a circular import, so we need to use Optional
```

Comment:
I don't feel like this is correct - where did you read about this?

Reply:
I added a comment to explain why. Basically Ruff doesn't recognize private fields (with `_`) of Pydantic models and think types are needed at runtime. So it's either we ignore the Ruff error for the ResultManager import at the top or we use forward references.

Reply:
Removed in 3c3e8ee3af4ff3ea57210e75835ee19d27f7105b